### ISSUE TITLE
GPU memory usage keep increasing

### ISSUE THREAD

**Issue by zed0630** [liutinglt/CE2P#8]
I am trying to reproduce the experiments result, but whenever I run the script, the gpu memory usage keeps increasing and the program frozen once it runs out of gpu memory after several iterations.
I have tried pytorch of different version, both 0.3.1 and 0.4.1 and the problem remained. The cuda version is 9.0 and the only modification I made is resizing the images to 473x473 in dataset.py due to [https://github.com/liutinglt/CE2P/issues/5](url). 
What is the possible reason that cause this strange behavior? Could you give the detailed settings of your experiments?

**Comment by liutinglt** [comment]
@zed0630 Sorry, there is a bug hasn't been fixed. Now, you can use the command ‘nvidia-smi topo -m’ to find the GPUDirect communication matrix like following. As our GPU 0-4 and 5-9 have the same connection PIX,  we use the 0-4 or 5-9 for training. You can check on your own machine, and make sure using the GPU with the same connection.  

------------------------------------------------------------------------------------------------------------- 
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    GPU8    GPU9    CPU Affinity
GPU0     X      PIX     PIX     PIX     PIX     PHB     PHB     PHB     PHB     PHB     0-9,20-29
GPU1    PIX      X      PIX     PIX     PIX     PHB     PHB     PHB     PHB     PHB     0-9,20-29
GPU2    PIX     PIX      X      PIX     PIX     PHB     PHB     PHB     PHB     PHB     0-9,20-29
GPU3    PIX     PIX     PIX      X      PIX     PHB     PHB     PHB     PHB     PHB     0-9,20-29
GPU4    PIX     PIX     PIX     PIX      X      PHB     PHB     PHB     PHB     PHB     0-9,20-29
GPU5    PHB     PHB     PHB     PHB     PHB      X      PIX     PIX     PIX     PIX     0-9,20-29
GPU6    PHB     PHB     PHB     PHB     PHB     PIX      X      PIX     PIX     PIX     0-9,20-29
GPU7    PHB     PHB     PHB     PHB     PHB     PIX     PIX      X      PIX     PIX     0-9,20-29
GPU8    PHB     PHB     PHB     PHB     PHB     PIX     PIX     PIX      X      PIX     0-9,20-29
GPU9    PHB     PHB     PHB     PHB     PHB     PIX     PIX     PIX     PIX      X      0-9,20-29

**Comment by zed0630** [comment]
@liutinglt Thank you. Now I can run the code with 4 GPUs, but I have to modify the batch size to 32 due to the limitation of gpu memory.

**Comment by rxqy** [comment]
Hi @zed0630 , I followed your advice and resize the images and labels to (473,473) in dataloader.
How about the performance? Can you achieve a similar result when you set batch to 32?

**Comment by zed0630** [comment]
Hi, @rxqy . I think I have reproduced a comparable result by following the instructions in the repo. There are only two differences in my experiments. The batch size is reduced to 32 due to limited gpu memory, and I generate the edge label using my own implementation rather than download the files in the repo.

**Comment by zed0630** [comment]
@rxqy You're right. Edge labels have the same shape with original images, and I resized those labels to 473x473 in training.
